### PR TITLE
[DAEMON-241] Fix telemetry guid and add deployType to events

### PR DESCRIPTION
**Pull request by ewanharris**
https://jira.appcelerator.org/browse/DAEMON-241

- Remove prod and preprod confs, merging prod into default
- Add deployType to events, setting to production if it doesn't exist
- Make gulp watch set deployType to development

### CODE REVIEW

File: docs/Getting-Started.md [hunk 1]
```diff
@@ -159,6 +159,8 @@ will rebuild that package and all parent packages, then restart the Appc Daemon.
 > malformed JavaScript code, it's likely going to cause `gulp` to exit, but the last spawned Appc
 > Daemon process will remain running. You may need to run `appcd stop` or `killall appcd`.
 
+When running the Appc Daemon with the `gulp watch` task telemetry will be sent using the `development` deployType.
```
Comment (cb1kenobi) [inline docs/Getting-Started.md]: Wrap at 100 chars.

File: packages/appcd-core/conf/default.js [hunk 2]
```diff
@@ -1,4 +1,8 @@
 module.exports = {
+	environment: {
```
Comment (cb1kenobi) [inline packages/appcd-core/conf/default.js]: This goes away.

File: gulpfile.js [hunk 3]
```diff
@@ -277,7 +277,7 @@ function runTests(cover, cb) {
  * watch/debug tasks
  */
 function startDaemon() {
-	spawn(process.execPath, [ 'packages/appcd/bin/appcd', 'start', '--debug' ], { stdio: 'inherit' });
+	spawn(process.execPath, [ 'packages/appcd/bin/appcd', 'start', '--debug', '--config', '{ \"telemetry\": { \"deployType\": \"development\" } }' ], { stdio: 'inherit' });
```
Comment (cb1kenobi) [inline gulpfile.js]: s/deployType/environment/

File: packages/appcd-core/conf/default.js [hunk 4]
```diff
@@ -145,6 +149,12 @@ module.exports = {
 	},
 
 	telemetry: {
+		/**
+		 * Deploy type for the analytics events.
+		 * @type {String}
+		 */
+		deployType: 'production',
```
Comment (cb1kenobi) [inline packages/appcd-core/conf/default.js]: s/deployType/environment/

File: packages/appcd-telemetry/src/telemetry.js [hunk 5]
```diff
@@ -64,6 +64,12 @@ export default class Telemetry extends Dispatcher {
 			url:           null
 		};
 
+		/**
+		 * The deploy type for the events.
+		 * @type {String}
+		 */
+		this.deployType = cfg.get('telemetry.deployType') || 'production';
```
Comment (cb1kenobi) [inline packages/appcd-telemetry/src/telemetry.js]: this.deployType = cfg.get('telemetry.environment') || 'production';

File: packages/appcd-core/conf/default.js (hunk 2 above)
Comment (cb1kenobi) [inline packages/appcd-core/conf/default.js]: Alphabetical order please.

**Comment by ewanharris** [comment]
@cb1kenobi Updated

**Pull request merged**